Products/CMFPlone/i18nl10n.py

```diff
@@ -2,49 +2,52 @@
 """
 Collection of i18n and l10n utility methods.
 """
-import re
-import logging
-
-from zope.component import getUtility
-from zope.i18n import translate
-from zope.i18n.locales import locales
-from zope.publisher.interfaces.browser import IBrowserRequest
-
 from Acquisition import aq_acquire
 from DateTime import DateTime
 from DateTime.interfaces import IDateTime
-
 from plone.registry.interfaces import IRegistry
 from Products.CMFPlone.utils import log
+from zope.component import getUtility
+from zope.i18n import translate
+from zope.i18n.locales import locales
+from zope.publisher.interfaces.browser import IBrowserRequest
 
-# these are taken from PTS, used for format interpolation
-NAME_RE = r"[a-zA-Z][a-zA-Z0-9_]*"
-_interp_regex = re.compile(r'(?<!\$)(\$(?:%(n)s|{%(n)s}))' % ({'n': NAME_RE}))
-
-datetime_formatvariables = ('H', 'I', 'm', 'd', 'M', 'p', 'S', 'Y', 'y', 'Z')
-name_formatvariables = ('a', 'A', 'b', 'B')
-
-ENGLISH_NAMES = {}
-try:
-    from DateTime.DateTime import _DAYS
-    ENGLISH_NAMES['_days'] = _DAYS
-    from DateTime.DateTime import _DAYS_A
-    ENGLISH_NAMES['_days_a'] = _DAYS_A
-    from DateTime.DateTime import _DAYS_P
-    ENGLISH_NAMES['_days_p'] = _DAYS_P
-    from DateTime.DateTime import _MONTHS
-    ENGLISH_NAMES['_months'] = _MONTHS
-    from DateTime.DateTime import _MONTHS_A
-    ENGLISH_NAMES['_months_a'] = _MONTHS_A
-    from DateTime.DateTime import _MONTHS_P
-    ENGLISH_NAMES['_months_p'] = _MONTHS_P
-except ImportError:
-    ENGLISH_NAMES['_days'] = DateTime._days
-    ENGLISH_NAMES['_days_a'] = DateTime._days_a
-    ENGLISH_NAMES['_days_p'] = DateTime._days_p
-    ENGLISH_NAMES['_months'] = DateTime._months
-    ENGLISH_NAMES['_months_a'] = DateTime._months_a
-    ENGLISH_NAMES['_months_p'] = DateTime._months_p
+import logging
+import re
+
+datetime_formatvariables = {'H', 'I', 'm', 'd', 'M', 'p', 'S', 'Y', 'y', 'Z'}
+name_formatvariables = {'a', 'A', 'b', 'B'}
+all_formatvariables = datetime_formatvariables | name_formatvariables
+_all_regexp_set = ','.join(all_formatvariables)
+# regexp to split up ${X} format strings
+_interp_regex = re.compile(
+    r'(?<!\$)(\$(?:[%(n)s]|{[%(n)s]}))' % ({'n': _all_regexp_set})
+)
+# regexp to detect if this is a strftime format string
+_dt_format_string_regexp = re.compile(r'\%([{0}])'.format(_all_regexp_set))
```

Comment:
It would be nice to have a very quick unittest to check this regexp

Reply:
i added two simple tests

Reply:
Thanks!